### PR TITLE
CLOUDP-242668: mocking tests + documentation changes

### PR DESCRIPTION
**Pull request by wtrocki**
## Description

Added mocking tests + documentation changes covering mocking

Link to any related issue(s):  CLOUDP-242668

### CODE REVIEW

File: docs/doc_2_error_handling.md [hunk 1]
```diff
@@ -42,3 +42,21 @@ if ok && apiError.GetError() == 404 {
  // Do something
 }
 ```
+
+## Mocking Errors
+
+SDK errors can be mocked by creating instance of `GenericOpenAPIError` struct. 
+
+```go
```
Comment (gssbzn) [inline docs/doc_2_error_handling.md:50]: let's add the same comment on surrounding code omitted ;)

File: examples/mock/cluster_test.go [hunk 2]
```diff
@@ -11,6 +11,15 @@ import (
 	"go.mongodb.org/atlas-sdk/v20240530001/mockadmin"
 )
 
+
+func MyFunctionCallingListClusters(clusterAPI admin.ClustersApi) (int, error) {
```
Comment (gssbzn) [inline examples/mock/cluster_test.go:15]:
```suggestion
func myFunctionCallingListClusters(clusterAPI admin.ClustersApi) (int, error) {
```

File: docs/doc_2_error_handling.md (hunk 1 above)
Comment (wtrocki) [reply docs/doc_2_error_handling.md:50]: `// Surrounding code omitted for brevity` - I see now. Adding

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**